### PR TITLE
verilator & systemc: update & integration

### PR DESCRIPTION
**Pull request by amiloradovsky**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/systemc/default.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "0sj8wlkp68cjhmkd9c9lvm3lk3sckczpz7w9vby64inc1f9fnf0b";
   };
 
+  enableParallelBuilding = true;
+  buildInputs = [ ];
+  nativeBuildInputs = [ autoconf automake libtool ];
+
+  preConfigure = "autoreconf -vi";
```
Comment (NickCao) [inline pkgs/applications/science/electronics/systemc/default.nix:18]: Or just use `autoreconfHook`?

File: pkgs/applications/science/electronics/systemc/default.nix [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-    sha256 = "5781b9a351e5afedabc37d145e5f7edec08f3fd5de00ffeb8fa1f3086b1f7b3f";
+  src = fetchFromGitHub {
+    owner = "accellera-official";
+    repo = pname;
+    rev = version;
+    sha256 = "0sj8wlkp68cjhmkd9c9lvm3lk3sckczpz7w9vby64inc1f9fnf0b";
   };
 
+  enableParallelBuilding = true;
+  buildInputs = [ ];
```
Comment (NickCao) [inline pkgs/applications/science/electronics/systemc/default.nix:15]:
```suggestion
```

**Comment by victormignot** [comment]
Result of  `nixpkgs-review pr 244644` on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>4 packages built:</summary>
fusesoc fusesoc.dist systemc verilator
</details>

**Review by ghost — APPROVED** [review]
Fine by me for the `systemc` side !

**Review by thoughtpolice — APPROVED** [review]
lgtm pending @NickCao's comments, though I think verilator could use some further clean up, but I can do that later.

**Pull request merged**